riscv/processor.cc

```diff
@@ -395,9 +395,8 @@ reg_t processor_t::vectorUnit_t::set_vl(int rd, int rs1, reg_t reqVL, reg_t newT
     vma = extract64(newType, 7, 1);
     vediv = 1 << extract64(newType, 8, 2);
 
-    vill = !(vflmul >= 0.125 && vflmul <= 8)
-           || vsew > ELEN
-           || vflmul < ((float)vsew_min / ELEN)
+    vill = vflmul > 8
+           || vsew > std::min(vflmul, 1.0f) * ELEN
```

Comment:
I think you need to restore the `vfmul >= 0.125` clause.  Once Spike supports ELEN=128, this check is not sufficient, because it would permit SEW=8 and LMUL=1/16, which is not a valid setting.

Reply:
The  PR has been updated.